src/functions/public/Get-ContextVault.ps1

```diff
@@ -0,0 +1,108 @@
+﻿function Get-ContextVault {
+    <#
+        .SYNOPSIS
+        Lists all available context vaults.
+
+        .DESCRIPTION
+        Retrieves information about all configured context vaults,
+        including their names, paths, and creation dates.
+        Shows which vault is currently set as the default.
+
+        .PARAMETER Name
+        Optional name filter to get information about a specific vault.
+
+        .EXAMPLE
+        Get-ContextVault
+
+        Output:
+        ```powershell
+        Name         Path                          Created              IsDefault
+        ----         ----                          -------              ---------
+        WorkVault    C:\Users\User\.contextvaults\WorkVault    2024-01-15 10:30:00 AM    True
+        PersonalVault C:\Users\User\.contextvaults\PersonalVault 2024-01-15 11:15:00 AM    False
+        ```
+
+        Lists all available context vaults.
+
+        .EXAMPLE
+        Get-ContextVault -Name "WorkVault"
+
+        Output:
+        ```powershell
+        Name         Path                          Created              IsDefault
+        ----         ----                          -------              ---------
+        WorkVault    C:\Users\User\.contextvaults\WorkVault    2024-01-15 10:30:00 AM    True
+        ```
+
+        Gets information about a specific vault.
+
+        .OUTPUTS
+        [PSCustomObject[]] Array of vault information objects.
+
+        .NOTES
+        Shows all registered vaults with their metadata and default status.
+
+        .LINK
+        https://psmodule.io/Context/Functions/Get-ContextVault/
```

Comment:
Think we should move the new functions in a folder called Vault. As its a new type of resource.